index.rst

```diff
@@ -1,59 +1,374 @@
+
+.. _collections-developer-processing-outputs:
+
+Developer processing outputs
+----------------------------
+
+Processing initiated by DM developers that are intended primarily for personal or small-group use must start with ``u/<user>`` (e.g. ``u/jbosch``), and are strongly encouraged to start with ``u/<user>/<ticket>`` (e.g. ``u/jbosch/DM-56789``) whenever possible.
+Names and structure after this prefix are at user discretion, but we strongly recommend using a combination of ``CHAINED`` collections and ``RUN`` collections to distinguish between "inputs and outputs" collections and "output only" collections, as in :ref:`collections-shared-official-processing-outputs`.
+The ``pipetask`` tool will automatically take care of this if the ``--output`` option is used with or instead of the ``--output-run`` option.
+
+.. note::
+   **TODO**: It's unclear whether BPS supports this currently, but it should be easy to at least support it under the condition that the ``RUN`` collection be given explicitly as well, instead of generated automatically by appending a timestamp.
+
+
+.. _collections-calibration-production:
+
+Calibration production
+----------------------
+
+Calibration production runs intended for broad use (i.e. outputs will be at least candidates for membership in the recommended calibration collection for this instrument) should output to collections with names that start with ``<instrument>/calib/<ticket>/``.
+Those produced for experimental or development purposes should start with ``u/<user>/<ticket>/``.
+
+In either case, the ``RUN`` collections that hold output datasets directly will usually require another disambiguating term, mapping roughly to the expected validity range epoch.
+Actual validity ranges are not assigned until datasets are certified (i.e. added to ``CALIBRATION`` collections), and until then, the usual dataset type + data ID constraint applies (i.e. there can only be one ``bias`` for each detector in a particular ``RUN`` collection).
+
+.. note::
+
+   **TODO**: We should resolve this uncertainty about what the last term should be in calibration production collections by the end of the RFC discussion period.
+   Gen2 used a "CalibDate", which I have always found a bit vague.
+   Some hash of input IDs seems a bit better, but it would need to be computed by external code, because we need the output collection name before we start processing (and also because the user probably wants it in some file before they launch any jobs, so they can easily look up what hashes mean).
+   Either should probably be combined with the timestamp suffixes that ``pipetask`` can automatically add to avoid clashes (especially differences  due to e.g. configuration rather than inputs).
+
+As noted in :ref:`collections-per-instrument`, certified calibration products intended for broad use should go in ``CALIBRATION`` collections named *just* ``<instrument>/calib/<ticket>``.
```

Comment:
Following the text, the output products from each of a ticket's calibration production RUN collections would then be certified into that ticket's final CALIBRATION collection.  Will the final \<instrument>/calib be a chained collection of these ticketed collection?  If so, then this seems like observed changes in the telescope only require new tickets to generate new calibrations after the estimated validStart date, which can then join the chained \<instrument>/calib.  I believe this system also works with algorithmic improvements: using the existing validStart values, we can regenerate updated calibrations for each date range.

Reply:
I would prefer to make `<instrument>/calib` a CHAINED collection that points to just one CALIBRATION collection that contains a version of everything, e.g. by having a new ticket's CALIBRATION collection copy certifications (but not datasets) from whatever was considered the previous best CALIBRATION collection.  That will rigorously enforce the constraint on one dataset per data ID and dataset type at any point in time, and it should make it a lot easier to generate fast queries for calibration lookups in `QuantumGraph` generation.